File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        
        vector<int> ans(queries.size());
        
        int sum = 0;
        
        for(int i : nums)
        {
            if (i%2 == 0)
            {
                sum += i;
            }
        }
        
        for(int q = 0; q < queries.size(); q++)
        {
            int val = queries[q][0];
            int ind = queries[q][1];
            
            if((nums[ind] + val)%2 == 0)
            {
                if(nums[ind]%2 != 0)
                {
                    sum += (nums[ind] + val);
                }
                
                else
                {
                    sum += val;
                }
                
                ans[q] = sum;
                nums[ind] += val;
                continue;
            }
            
            if(nums[ind]%2 == 0)
            {
                sum -= nums[ind];
                
                
            }
            ans[q] = sum;
            nums[ind] += val;
            
            
            
            
        }
        
        
        return ans;
    }
};

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        
        evenSum = sum(list(filter(lambda x : x%2 == 0, nums)))
        ans = [0] * len(queries)
        
        for index, (val, ind) in enumerate(queries):
            
            if (val + nums[ind])%2 == 0:
                
                if nums[ind]%2 == 0:
                    
                    evenSum += val
                else:
                    evenSum += (nums[ind] + val)
            else:
                
                if nums[ind]%2 == 0:
                    evenSum -= nums[ind]
            
            nums[ind] += val
            ans[index] = evenSum
        
        return ans
        